src/scenes/Event/store.js

```diff
@@ -69,25 +93,54 @@ export default class {
     if (this.unconfirmed) return _.find(this.oracles, { txid: this.txid });
     return _.find(this.oracles, { address: this.address }) || {};
   }
+  // both
   @computed get selectedOption() {
-    return this.oracle.options[this.selectedOptionIdx] || {};
+    return this.event.options[this.selectedOptionIdx] || {};
+  }
+  @computed get event() {
+    if (this.type === 'topic') return this.topic;
+    return this.oracle;
   }
 
   constructor(app) {
     this.app = app;
   }
 
-  // TODO: go from /oracle/null/null/:txid -> /oracle/:topicAddress/:address/:txid
-  routeToConfirmedOracle = () => {
-    // const { topicAddress, address, txid } = this;
-    // this.app.router.push(`/oracle/${topicAddress}/${address}/${txid}`);
-  }
-
   @action
-  async init({ topicAddress, address, txid }) {
+  async init({ topicAddress, address, txid, type }) {
+    this.reset();
     this.topicAddress = topicAddress;
     this.address = address;
     this.txid = txid;
+    this.type = type;
+    if (type === 'topic') {
+      // GraphQL calls
+      this.escrowAmount = await this.getEscrowAmount();
+      const topics = await queryAllTopics([{ address }], undefined, 1, 0);
+      const transactions = await queryAllTransactions([{ topicAddress: address }], { field: 'createdTime', direction: SortBy.DESCENDING });
+
+      // API calls
+      const { bets, votes } = await this.getBetAndVoteBalances();
+      const withdrawableAddresses = await this.getWithdrawableAddresses();
+      const oracles = await this.getAllOracles(address);
+      // console.log('ADDRESSES: ', withdrawableAddresses);
+      runInAction(() => {
+        // this.escrowAmount = satoshiToDecimal(escrowRes.data.result[0]);
```

Comment:
This should be used. We want to convert the `escrowAmount` to decimal format instead of Botoshi which is what we receive it in. 

1 BOT = 100000000 Botoshi

Reply:
updated